src/tblite/coulomb/thirdorder.f90

```
@@ -170,6 +172,33 @@ subroutine get_potential(self, mol, cache, wfn, pot)
 end subroutine get_potential
 
 
+!> Evaluate gradient of charge dependent potential shift from the interaction
+subroutine get_potential_gradient(self, mol, cache, wfn, pot)
+   !> Instance of the electrostatic container
+   class(onsite_thirdorder), intent(in) :: self
+   !> Molecular structure data
+   type(structure_type), intent(in) :: mol
+   !> Reusable data container
+   type(container_cache), intent(inout) :: cache
+   !> Wavefunction data
+   type(wavefunction_type), intent(in) :: wfn
+   !> Density dependent potential
+   type(potential_type), intent(inout) :: pot
+
+   integer :: iat, izp, ii, ish
+
+   if (.not. self%shell_resolved) then
```

What happens in the shell resolved case? Can we at least raise an error if this is not supported rather than falling through silently

---

You are right that it doesn't make much sense to break the support for both shell- and atom-resolved ES over this small addition. I added the shell-resolved case and tested it in the CEH context with atomic charges distributed over the shells.